### PR TITLE
Add tests for preload-file and locateFile

### PR DESCRIPTION
**Pull request by askuri**
For preload-file:
Test that putting .js, .wasm and .data files in a sub-directory
below the .html file still works. As stated in the reference,
emscripten should expect to find .data and .wasm files next to
the .js file.

For locateFile:
Test locateFile when using preload-file instead of manually
running the file packager.

See issue #9637

### CODE REVIEW

File: tests/test_browser.py [hunk 1]
```diff
@@ -260,12 +260,36 @@ def make_main(path):
...
+    # used for test cases, where .js, .wasm and .data files are in a subdirectory
+    ensure_dir('sub')
+    create_test_file(os.path.join('sub', 'page_sub.html'), '''
+      <!DOCTYPE html>
+      <html lang="en">
+        <head>
+          <script src="sub2/page.js"></script>
+          <script>
+            Module['onRuntimeInitialized'] = function() {
+              //new Module.page();
```
Comment (kripken) [inline tests/test_browser.py:273]: is this line needed? (if not, the entire script tag seems unnecessary?)

Comment (askuri) [reply tests/test_browser.py]: Indeed not necessary, I removed the whole second script tag. Thanks for the suggestion.

**Comment by kripken** [comment]
Thanks @askuri !

For the first issue, we should probably update the docs. This is indeed maybe a little odd, but it's been the behavior to not use the script dir for this for many years now, so it could be a breaking change for people. But we should definitely document it properly.

For the second issue the problem is that `--preload-file` puts the file preloading code *before* the `--pre-js` which adds `locateFile`. There are definitely tests that depend on that order, so it also may not be worth changing, and just documenting. However, it may also be worth looking into changing that order.

If you're interested to update the docs for these that would be great! Can use this same PR if that's convenient. (If not, we should probably file separate issues for the two problems here.)

**Comment by askuri** [comment]
"For the second issue the problem is that --preload-file puts the file preloading code before the --pre-js which adds locateFile."
If I understand you correctly, that means that currently there is no way to have the data file at another location than next to the HTML file? That would be unfortunate, I guess. What about people who'd like to host it on another server like they probably already do with other assets?

I'm sorry I won't have time to contribute / update the docs at least until July 10th.

**Comment by kripken** [comment]
> If I understand you correctly, that means that currently there is no way to have the data file at another location than next to the HTML file? That would be unfortunate, I guess. What about people who'd like to host it on another server like they probably already do with other assets?

Yes, I agree that is a limitation that isn't great. It would be good to think about how this could be improved in a backwards-compatible manner. For example maybe we could check locateFile asynchronously (as file preloading is async anyhow) and then that would happen at a good time when locateFile is there.

> I'm sorry I won't have time to contribute / update the docs at least until July 10th.

No rush! Would be great to see progress on this whenever you have time.

**Pull request closed without merging**